### ISSUE TITLE
Add indicator for changesets merged but not targeted by current spec

### ISSUE THREAD

**Issue by BolajiOlajide** [sourcegraph/sourcegraph#49880]

**Comment by courier-new** [comment]
We've added this as a design session topic but haven't gotten to it yet.

**Comment by BolajiOlajide** [comment]
Tooltip text: `The Batch Spec was updated after merging this changeset. Therefore it is targeted by a previous spec.`

**Comment by BolajiOlajide** [comment]
<img width="547" alt="CleanShot 2023-04-13 at 17 17 06@2x" src="https://user-images.githubusercontent.com/25608335/231821954-e329649b-a01a-4444-a593-6961e4f991fa.png">

**Comment by courier-new** [comment]
Are we super attached to the wording "using old spec" here? "Using" in particular makes it sound like the spec somehow did the merging, so maybe we could say "from" instead? I also don't love old as the adjective lol, what about "previous"?

**Merged**
from previous spec

**Comment by courier-new** [comment]
I don't think this information is readily available over GraphQL yet. We would want to know the ID of the batch spec that produced each changeset and compare that to the latest applied batch spec for the batch change. We should probably surface the `BatchSpecID` on the `VisibleChangesetSpec` so that it could be queried like so:

```graphql
.. on ExternalChangeset {
  id
  currentSpec {
    batchSpecID
  }
}
```

I would personally estimate 0.5d for the backend and 0.5d for the frontend in that case.
